### PR TITLE
Pagination bar

### PR DESCRIPTION
**Pull request by MI6-451**
## Summary of the changes
Adds the pagination bar component which houses a pagination component with additional controls and information displayed. Some aspects of the design have not yet been implemented, associated tickets/issues are linked below. 

## Related issue
Component ticket/issue: #460 
Blocked by: #611, #612, #613, 

## Checklist
- [x] I have added relevant unit and visual regression tests.
- [x] I have manually accessibility tested any changes, if relevant.
- [x] I have ensured any changes match the Figma component library.

### CODE REVIEW

**Comment by jd239** [comment]
Is it worth closing this now and then reopening when all the blockers have been completed?

File: packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx
Comment (GCHQ-Developer-847) [inline packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:1]: The 'go to page' input shows the error validation state when you first tab to it and says 'Please enter a number'

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]
Looks great :) only a few small comments from me

File: packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx
Comment (GCHQ-Developer-530) [reply packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:1]: This also happens if you input a number and press enter

Comment (MI6-451) [reply packages/web-components/src/components/ic-pagination-bar/ic-pagination-bar.tsx:1]: Should both be sorted now

**Pull request merged**